### DISCUSSION TITLE
Qt-DAB versions 4 and 5

### DISCUSSION THREAD

**Discussion by JvanKatwijk** [JvanKatwijk/qt-dab#269] [Announcements]
An old plan was to upgrade the dabMini program, just an interface as simple as possible, just for listening to a service
Implementation of that plan ended up in a new GUI for Qt-DAB, and a version next tot 4.4

While called V5, it is functionally the same as the 4.4 version, just with a new front. Basically almost all controls are shifted to the
configuration window (which is called configuration and control window), and just a few buttons for interacting are left on the
main widget.
So, V5 is not a replacemement, and updates to the 4 version will - at least for now - be numbered 4.x.x

To be complete, I generated AppImages and Windows installers for both the 4 and 5 version.

I run under Windows the 4 and 5 versions next to each other, they have different identifications

![qt-dab-screen](https://user-images.githubusercontent.com/8713152/176134492-a102efbd-ea32-4bf6-8d0b-4abebf5152a1.png
![qt-dab-5](https://user-images.githubusercontent.com/8713152/182400335-71ed8cfc-cc06-44f5-a5ab-b26c52fa4c46.png)
![qt-dab-4b](https://user-images.githubusercontent.com/8713152/182400364-ff440777-1cf2-42a4-a582-11167ed8b807.png)
)

<hr /><em>This discussion was created from the release <a href='https://github.com/JvanKatwijk/qt-dab/releases/tag/'>Qt-DAB versions 4 and 5</a>.</em>

**Comment by DG9BFC** [comment]
hmm no plutotx in version 5.0 ??

**Comment by JvanKatwijk** [comment]
No, not in the precompiled versions. You can configure it when creating
your own executable


Op do 11 aug. 2022 om 20:03 schreef DG9BFC ***@***.***>:

> hmm no plutotx in version 5.0 ??
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-3377922>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQCXXUVEDO6ZLCOG5XDVYU565ANCNFSM55LNPM2Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Comment by markov035** [comment]
With v4 or5, I get this error, something is hanging in my config?
(x-ubuntu)

Buffersize = 5, filterSize 5
Could not find rtlsdr_set_tuner_bandwidth
error = 21


but it is connected:
lsusb gives ... Bus 003 Device 003: ID 0bda:2838 Realtek Semiconductor Corp. RTL2838 DVB-T

Thanks,
Marc

**Comment by JvanKatwijk** [comment]
Your message seems to show that the rtlsdr library you have installed does
not contain the "set_tuner_bandwidth" function",
must be an older version.
To remedy that I uploaded a version of Qt-DAB-5 appimage where the function
is not used if not available
You could give that a try, if it works I'll adapt the Qt-DAB-4 appimage as
well


Op zo 12 mrt 2023 om 00:01 schreef markov035 ***@***.***>:

> With v4 or5, I get this error, something is hanging in my config?
> (x-ubuntu)
>
> Buffersize = 5, filterSize 5
> Could not find rtlsdr_set_tuner_bandwidth
> error = 21
>
> but it is connected:
> lsusb gives ... Bus 003 Device 003: ID 0bda:2838 Realtek Semiconductor
> Corp. RTL2838 DVB-T
>
> Thanks,
> Marc
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-5278384>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQDHYLWPOC2AXMTOF2LW3T74PANCNFSM55LNPM2Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Comment by markov035** [comment]
Op 12/03/2023 om 09:54 schreef JvanKatwijk:
> Your message seems to show that the rtlsdr library you have installed does
> not contain the "set_tuner_bandwidth" function",
> must be an older version.


Hallo, ik spreek Nederlands!

Strange, I get an error that seems to explain more ...


=====



Kernel driver is active, or device is claimed by second instance of 
librtlsdr.
In the first case, please either detach or blacklist the kernel module
(dvb_usb_rtl28xxu), or enable automatic detaching at compile time.

usb_claim_interface error -6
Opening rtlsdr device failed
error = 23


======




Marc

**Comment by JvanKatwijk** [comment]
De "gewone" rtlsdr driver die uit de ubuntu repository komt is fout
geconfigureerd, de kernel detach option
is niet aangezet. Dat betekent dat na een keer sluiten de driver niet
opnieuw kan worden geladen


Op zo 12 mrt 2023 om 15:35 schreef markov035 ***@***.***>:

> Op 12/03/2023 om 09:54 schreef JvanKatwijk:
> > Your message seems to show that the rtlsdr library you have installed
> does
> > not contain the "set_tuner_bandwidth" function",
> > must be an older version.
>
>
> Hallo, ik spreek Nederlands!
>
> Strange, I get an error that seems to explain more ...
>
>
> =====
>
>
>
> Kernel driver is active, or device is claimed by second instance of
> librtlsdr.
> In the first case, please either detach or blacklist the kernel module
> (dvb_usb_rtl28xxu), or enable automatic detaching at compile time.
>
> usb_claim_interface error -6
> Opening rtlsdr device failed
> error = 23
>
>
> ======
>
>
>
>
> Marc
>
> > To remedy that I uploaded a version of Qt-DAB-5 appimage where the
> > function
> > is not used if not available
> > You could give that a try, if it works I'll adapt the Qt-DAB-4 appimage
> as
> > well
> >
> >
> > Op zo 12 mrt 2023 om 00:01 schreef markov035 ***@***.***>:
> >
> > > With v4 or5, I get this error, something is hanging in my config?
> > > (x-ubuntu)
> > >
> > > Buffersize = 5, filterSize 5
> > > Could not find rtlsdr_set_tuner_bandwidth
> > > error = 21
> > >
> > > but it is connected:
> > > lsusb gives ... Bus 003 Device 003: ID 0bda:2838 Realtek Semiconductor
> > > Corp. RTL2838 DVB-T
> > >
> > > Thanks,
> > > Marc
> > >
> > > —
> > > Reply to this email directly, view it on GitHub
> > >
> > <
> https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-5278384
> >,
> > > or unsubscribe
> > >
> > <
> https://github.com/notifications/unsubscribe-auth/ACCPHQDHYLWPOC2AXMTOF2LW3T74PANCNFSM55LNPM2Q
> >
> > > .
> > > You are receiving this because you authored the thread.Message ID:
> > > ***@***.***>
> > >
> >
> >
> > --
> > Jan van Katwijk
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-5280014>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/A6NYUG7DX3GJULM62PAUZZ3W3WFLPANCNFSM55LNPM2Q
> >.
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-5281606>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQB2ASNJQK4YES2ML6TW3XNM5ANCNFSM55LNPM2Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Comment by JvanKatwijk** [comment]
Je kan de kernel module blacklisten, of je kan zelf een librtlsdr.so maken,
dat laatste is vrij eenvoudig, kijk op
de osmocom site


Op zo 12 mrt 2023 om 15:35 schreef markov035 ***@***.***>:

> Op 12/03/2023 om 09:54 schreef JvanKatwijk:
> > Your message seems to show that the rtlsdr library you have installed
> does
> > not contain the "set_tuner_bandwidth" function",
> > must be an older version.
>
>
> Hallo, ik spreek Nederlands!
>
> Strange, I get an error that seems to explain more ...
>
>
> =====
>
>
>
> Kernel driver is active, or device is claimed by second instance of
> librtlsdr.
> In the first case, please either detach or blacklist the kernel module
> (dvb_usb_rtl28xxu), or enable automatic detaching at compile time.
>
> usb_claim_interface error -6
> Opening rtlsdr device failed
> error = 23
>
>
> ======
>
>
>
>
> Marc
>
> > To remedy that I uploaded a version of Qt-DAB-5 appimage where the
> > function
> > is not used if not available
> > You could give that a try, if it works I'll adapt the Qt-DAB-4 appimage
> as
> > well
> >
> >
> > Op zo 12 mrt 2023 om 00:01 schreef markov035 ***@***.***>:
> >
> > > With v4 or5, I get this error, something is hanging in my config?
> > > (x-ubuntu)
> > >
> > > Buffersize = 5, filterSize 5
> > > Could not find rtlsdr_set_tuner_bandwidth
> > > error = 21
> > >
> > > but it is connected:
> > > lsusb gives ... Bus 003 Device 003: ID 0bda:2838 Realtek Semiconductor
> > > Corp. RTL2838 DVB-T
> > >
> > > Thanks,
> > > Marc
> > >
> > > —
> > > Reply to this email directly, view it on GitHub
> > >
> > <
> https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-5278384
> >,
> > > or unsubscribe
> > >
> > <
> https://github.com/notifications/unsubscribe-auth/ACCPHQDHYLWPOC2AXMTOF2LW3T74PANCNFSM55LNPM2Q
> >
> > > .
> > > You are receiving this because you authored the thread.Message ID:
> > > ***@***.***>
> > >
> >
> >
> > --
> > Jan van Katwijk
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-5280014>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/A6NYUG7DX3GJULM62PAUZZ3W3WFLPANCNFSM55LNPM2Q
> >.
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-5281606>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQB2ASNJQK4YES2ML6TW3XNM5ANCNFSM55LNPM2Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Comment by DG9BFC** [comment]
fer ent fiwe werked all heer :-)
dg9bfc
ik spreek op plaat dat versteid ji doch

**Comment by DG9BFC** [comment]
dat wered precompilired windows versie fer ent fiwe

**Comment by ArieGroeneveld** [comment]
"Since there seemed to be some problems in fetching the library for the 3.10 API, the code was slightly adapted to deal with that
so the installer here should work with the 3.10 API"

In mijn geval werkt het niet met de 3.10 API (can't fetch the library) dus ik heb API 3.09 weer geïnstalleerd.

**Comment by JvanKatwijk** [comment]
Bijzonder, staat de sdrplay_api.dll in  the program files folder als
C:\Program Files\SDRplay\API\x86\sdrplay_api.dll ?

Op vr 30 jun 2023 om 12:54 schreef ArieGroeneveld ***@***.***
>:

> "Since there seemed to be some problems in fetching the library for the
> 3.10 API, the code was slightly adapted to deal with that
> so the installer here should work with the 3.10 API"
>
> In mijn geval werkt het niet met de 3.10 API (can't fetch the library) dus
> ik heb API 3.09 weer geïnstalleerd.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-6324749>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQBDQN2JBVPGHWJ23RTXN2V5LANCNFSM55LNPM2Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Reply by ArieGroeneveld** [reply]
Jazeker. Een sdrplay_api.dll en een sdrplay_api.lib

**Comment by JvanKatwijk** [comment]
De default manier om libraries te halen is opm via de registry, als dat
niet lukt wordt er een functie aangeroepen die heel expliciet
de sdrplay_api.dll in de genoemde folder probeert te openen.
Als dat niet lukt is er dus "iets" aan de hand met de instellingen, dan
"mag" de "LoadLibrary" functie kennelijk niet in
genoemde folder komen of mag de sdrplay_api.dll niet lezen.

IK heb geen problemen ermee, in W10 draai ik een API 11, dus ik kan
momenteel hier niet veel mee





Op vr 30 jun 2023 om 14:52 schreef ArieGroeneveld ***@***.***
>:

> Jazeker. Een sdrplay_api.dll en een sdrplay_api.lib
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/269#discussioncomment-6325688>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQBIP64BOYNJZQYB6U3XN3D2JANCNFSM55LNPM2Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Reply by ArieGroeneveld** [reply]
Toch heeft je antwoord me geholpen. Als er ergens iets fout gaat is het misschien een goed idee om SDRUno opnieuw te installeren. Dat heb ik gedaan en het loopt als een zonnetje nu, met API 3.11. Bedankt!
